EVPN Ethernet Segment tests fail due to incorrect MAC address format in YAML definition

## Description

  The EVPN Ethernet Segment acceptance tests are failing in the CI pipeline on C9000v 17.15.1 due to a MAC
  address format mismatch. Tests are using colon-separated MAC address format (`00:11:22:33:44:55`) when the
  YANG model requires Cisco dotted notation (`0011.2233.4455`).

  ## Pipeline Failure Logs

```
  [2025-11-06T03:44:32.106Z] === RUN   TestAccDataSourceIosxeEVPNEthernetSegment
  [2025-11-06T03:44:32.407Z]     data_source_iosxe_evpn_ethernet_segment_test.go:45: Step 1/1 error: Error
  running pre-apply plan: exit status 1
  [2025-11-06T03:44:32.407Z][2025-11-06T03:44:32.407Z]         Error: Invalid Attribute Value Match
  [2025-11-06T03:44:32.408Z][2025-11-06T03:44:32.408Z]           with iosxe_evpn_ethernet_segment.test,
  [2025-11-06T03:44:32.408Z]           on terraform_plugin_test.tf line 11, in resource
  "iosxe_evpn_ethernet_segment" "test":
  [2025-11-06T03:44:32.408Z]           11: resource "iosxe_evpn_ethernet_segment" "test" {
  [2025-11-06T03:44:32.408Z][2025-11-06T03:44:32.408Z]         Attribute identifier_types[0].system_mac value
   must match regular expression
  [2025-11-06T03:44:32.408Z]         '[0-9a-fA-F]+[0-9a-fA-F]+[0-9a-fA-F]+', got: 00:11:22:33:44:55
  [2025-11-06T03:44:32.509Z] --- FAIL: TestAccDataSourceIosxeEVPNEthernetSegment (0.37s)
```

  ## Root Cause Analysis

  ### Issue Chain

  1. **YANG Model Pattern (Correct)** ✅
     - The `Cisco-IOS-XE-l2vpn` YANG model defines the `system-mac` leaf with pattern:
  `[0-9a-fA-F]+\.[0-9a-fA-F]+\.[0-9a-fA-F]+`
     - This expects Cisco dotted notation: `0011.2233.4455`

  2. **Provider Generator (Correct)** ✅
     - Generator correctly extracts pattern from YANG and applies it as schema validation
     - Generated code in `internal/provider/resource_iosxe_evpn_ethernet_segment.go:136`

  3. **YAML Definition (INCORRECT)** ❌
     - **File:** `gen/definitions/evpn_ethernet_segment.yaml:37`
     - **Current:** `example: 00:11:22:33:44:55` (colon-separated)
     - **Should be:** `example: 0011.2233.4455` (Cisco dotted)

  4. **Generated Tests (Incorrect due to YAML)** ❌
     - Test uses: `system_mac = "00:11:22:33:44:55"`
     - Validation fails before reaching device

  5. **Generated Documentation (Incorrect due to YAML)** ❌
     - Documentation examples show wrong format

  ### Format Comparison

```md
  | Component | Format | Example | Status |
  |-----------|--------|---------|--------|
  | YANG Model Pattern | Cisco Dotted | `0011.2233.4455` | ✅ Correct |
  | Provider Schema Validator | Cisco Dotted | Pattern from YANG | ✅ Correct |
  | YAML Example | Colon-Separated | `00:11:22:33:44:55` | ❌ Wrong |
  | Generated Test | Colon-Separated | `00:11:22:33:44:55` | ❌ Wrong |
  | Generated Docs | Colon-Separated | `00:11:22:33:44:55` | ❌ Wrong |
```

  ## Fix Required

  **File:** `gen/definitions/evpn_ethernet_segment.yaml`
  **Line:** 37

  Change:

  ```yaml
  - yang_name: esi-type-choice/system-mac/system-mac
    xpath: system-mac
    example: 00:11:22:33:44:55
```

  To:
```yaml
  - yang_name: esi-type-choice/system-mac/system-mac
    xpath: system-mac
    example: 0011.2233.4455
```

  Then regenerate code:
```
  make gen NAME="EVPN Ethernet Segment"
```

  This will update:
  - Test files with correct MAC format
  - Documentation examples
  - All generated code

  Related Information

  - Introduced in: PR #321  (commit 308d5ff)
  - Affected platforms: C9000v 17.15.1 (tests are C9000V-specific)
  - Test scope: Both TestAccDataSourceIosxeEVPNEthernetSegment and TestAccIosxeEVPNEthernetSegment
  - Note: The hex_string field (line 34) correctly uses Cisco dotted format: 0001.0000.0000.0000.000A

--- a/docs/resources/evpn_ethernet_segment.md
+++ b/docs/resources/evpn_ethernet_segment.md
@@ -20,9 +20,8 @@ resource "iosxe_evpn_ethernet_segment" "example" {
   redundancy_single_active = true
   identifier_types = [
     {
-      type       = 0
-      hex_string = "0001.0000.0000.0000.000A"
-      system_mac = "00:11:22:33:44:55"
+      type       = 3
+      system_mac = "0011.2233.4455"
     }
   ]
 }

--- a/examples/resources/iosxe_evpn_ethernet_segment/resource.tf
+++ b/examples/resources/iosxe_evpn_ethernet_segment/resource.tf
@@ -5,9 +5,8 @@ resource "iosxe_evpn_ethernet_segment" "example" {
   redundancy_single_active = true
   identifier_types = [
     {
-      type       = 0
-      hex_string = "0001.0000.0000.0000.000A"
-      system_mac = "00:11:22:33:44:55"
+      type       = 3
+      system_mac = "0011.2233.4455"
     }
   ]
 }

--- a/gen/definitions/evpn_ethernet_segment.yaml
+++ b/gen/definitions/evpn_ethernet_segment.yaml
@@ -27,11 +27,12 @@ attributes:
       - yang_name: identifier-type
         tf_name: type
         id: true
-        example: 0
+        example: 3
       - yang_name: esi-type-choice/hex-string/hex-string
         xpath: hex-string
         tf_name: hex_string
-        example: 0001.0000.0000.0000.000A
+        example: 00.00.00.00.00.00.01.01.01
+        exclude_test: true
       - yang_name: esi-type-choice/system-mac/system-mac
         xpath: system-mac
-        example: 00:11:22:33:44:55
+        example: 0011.2233.4455

--- a/internal/provider/data_source_iosxe_evpn_ethernet_segment_test.go
+++ b/internal/provider/data_source_iosxe_evpn_ethernet_segment_test.go
@@ -39,9 +39,8 @@ func TestAccDataSourceIosxeEVPNEthernetSegment(t *testing.T) {
 	checks = append(checks, resource.TestCheckResourceAttr("data.iosxe_evpn_ethernet_segment.test", "df_election_wait_time", "3"))
 	checks = append(checks, resource.TestCheckResourceAttr("data.iosxe_evpn_ethernet_segment.test", "redundancy_all_active", "false"))
 	checks = append(checks, resource.TestCheckResourceAttr("data.iosxe_evpn_ethernet_segment.test", "redundancy_single_active", "true"))
-	checks = append(checks, resource.TestCheckResourceAttr("data.iosxe_evpn_ethernet_segment.test", "identifier_types.0.type", "0"))
-	checks = append(checks, resource.TestCheckResourceAttr("data.iosxe_evpn_ethernet_segment.test", "identifier_types.0.hex_string", "0001.0000.0000.0000.000A"))
-	checks = append(checks, resource.TestCheckResourceAttr("data.iosxe_evpn_ethernet_segment.test", "identifier_types.0.system_mac", "00:11:22:33:44:55"))
+	checks = append(checks, resource.TestCheckResourceAttr("data.iosxe_evpn_ethernet_segment.test", "identifier_types.0.type", "3"))
+	checks = append(checks, resource.TestCheckResourceAttr("data.iosxe_evpn_ethernet_segment.test", "identifier_types.0.system_mac", "0011.2233.4455"))
 	resource.Test(t, resource.TestCase{
 		PreCheck:                 func() { testAccPreCheck(t) },
 		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
@@ -69,9 +68,8 @@ func testAccDataSourceIosxeEVPNEthernetSegmentConfig() string {
 	config += `	redundancy_all_active = false` + "\n"
 	config += `	redundancy_single_active = true` + "\n"
 	config += `	identifier_types = [{` + "\n"
-	config += `		type = 0` + "\n"
-	config += `		hex_string = "0001.0000.0000.0000.000A"` + "\n"
-	config += `		system_mac = "00:11:22:33:44:55"` + "\n"
+	config += `		type = 3` + "\n"
+	config += `		system_mac = "0011.2233.4455"` + "\n"
 	config += `	}]` + "\n"
 	config += `}` + "\n"
 

--- a/internal/provider/resource_iosxe_evpn_ethernet_segment_test.go
+++ b/internal/provider/resource_iosxe_evpn_ethernet_segment_test.go
@@ -42,9 +42,8 @@ func TestAccIosxeEVPNEthernetSegment(t *testing.T) {
 	checks = append(checks, resource.TestCheckResourceAttr("iosxe_evpn_ethernet_segment.test", "df_election_wait_time", "3"))
 	checks = append(checks, resource.TestCheckResourceAttr("iosxe_evpn_ethernet_segment.test", "redundancy_all_active", "false"))
 	checks = append(checks, resource.TestCheckResourceAttr("iosxe_evpn_ethernet_segment.test", "redundancy_single_active", "true"))
-	checks = append(checks, resource.TestCheckResourceAttr("iosxe_evpn_ethernet_segment.test", "identifier_types.0.type", "0"))
-	checks = append(checks, resource.TestCheckResourceAttr("iosxe_evpn_ethernet_segment.test", "identifier_types.0.hex_string", "0001.0000.0000.0000.000A"))
-	checks = append(checks, resource.TestCheckResourceAttr("iosxe_evpn_ethernet_segment.test", "identifier_types.0.system_mac", "00:11:22:33:44:55"))
+	checks = append(checks, resource.TestCheckResourceAttr("iosxe_evpn_ethernet_segment.test", "identifier_types.0.type", "3"))
+	checks = append(checks, resource.TestCheckResourceAttr("iosxe_evpn_ethernet_segment.test", "identifier_types.0.system_mac", "0011.2233.4455"))
 	resource.Test(t, resource.TestCase{
 		PreCheck:                 func() { testAccPreCheck(t) },
 		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
@@ -106,9 +105,8 @@ func testAccIosxeEVPNEthernetSegmentConfig_all() string {
 	config += `	redundancy_all_active = false` + "\n"
 	config += `	redundancy_single_active = true` + "\n"
 	config += `	identifier_types = [{` + "\n"
-	config += `		type = 0` + "\n"
-	config += `		hex_string = "0001.0000.0000.0000.000A"` + "\n"
-	config += `		system_mac = "00:11:22:33:44:55"` + "\n"
+	config += `		type = 3` + "\n"
+	config += `		system_mac = "0011.2233.4455"` + "\n"
 	config += `	}]` + "\n"
 	config += `}` + "\n"
 	return config

